lib/zendesk_api/search.rb

```diff
@@ -0,0 +1,51 @@
+# `zendesk_api` gem root
+module ZendeskAPI
+  # A rich factory that returns a class for your searches
+  class Search
+    # Creates a search collection
+    def self.search(client, options = {})
+      if (options.keys.map(&:to_s) & %w[query external_id]).empty?
+        warn "you have not specified a query for this search"
```

Comment:
What does this warn do? It is just a log from what I found online.

Reply:
If properly configured, I believe it can end up in Datadog too, but I didn't investigate much and usually I tend to a logger instance.

https://www.educative.io/edpresso/what-is-warn-method-in-rubys-kernel-object